plugins/events-backend-module-sqs/src/service/AwsSqsConsumingEventPublisher.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DeleteMessageBatchCommand,
+  Message,
+  ReceiveMessageCommand,
+  ReceiveMessageCommandInput,
+  SQSClient,
+} from '@aws-sdk/client-sqs';
+import { PluginTaskScheduler } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import { EventBus, EventPublisher } from '@backstage/plugin-events-node';
+import { Logger } from 'winston';
+import { AwsSqsEventSourceConfig, readConfig } from './config';
+
+/**
+ * Publishes events received from an AWS SQS queue.
+ * The message payload will be used as event payload and passed to registered subscribers.
+ *
+ * @public
+ */
+export class AwsSqsConsumingEventPublisher implements EventPublisher {
```

Comment:
this thing (and probably more of them) should totally be hooked up to some nice prom metrics later

Reply:
Do you have an example for that? Should it be part of this PR?

Reply:
Nope, later. Just noting. You'll see that catalog-backend and scaffolder-backend have some stuff with prom-client in them.

Reply:
I couldn't find anything at scaffolder-backend but at catalog-backend

Might make sense to expose https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/util/metrics.ts somewhere as general helpers 🤔 

I left TODOs for now so we can follow up on this.